openedx/core/djangoapps/credit/models.py

```diff
@@ -215,6 +228,25 @@ class CreditEligibility(TimeStampedModel):
     class Meta(object):  # pylint: disable=missing-docstring
         unique_together = ('username', 'course')
 
+    @classmethod
+    def get_user_eligibility(cls, username):
+        """
+        returns list of all eligible courses
+        """
+        return cls.objects.filter(username=username).select_related('course', 'course__providers')
+
+    @classmethod
+    def is_credit_course(cls, course_key, username):
+        """Check that given course is credit or not.
```

Comment:
This method is doing something different from what the name and documentation imply. It is returning an indication of the user's eligibility, not the course's credit status.